devnet-sdk/system2/cluster.go

```diff
@@ -0,0 +1,65 @@
+package system2
+
+import (
+	"github.com/ethereum-optimism/optimism/op-supervisor/supervisor/backend/depset"
+)
+
+// L2ClusterID identifies a Cluster by name, is type-safe, and can be value-copied and used as map key.
+type L2ClusterID genericID
+
+func (id L2ClusterID) String() string {
+	return genericID(id).string("L2Cluster")
+}
+
+func (id L2ClusterID) MarshalText() ([]byte, error) {
+	return genericID(id).marshalText("L2Cluster")
+}
+
+func (id *L2ClusterID) UnmarshalText(data []byte) error {
+	return (*genericID)(id).unmarshalText("L2Cluster", data)
+}
+
+func SortL2ClusterIDs(ids []L2ClusterID) []L2ClusterID {
+	return copyAndSortCmp(ids)
+}
+
+// Cluster represents a set of L2 chains that interop
+type Cluster interface {
+	Common
+	ID() L2ClusterID
+
+	DependencySet() depset.DependencySet
+}
+
+// ClusterConfig is the config to create a default cluster object
+type ClusterConfig struct {
+	CommonConfig
+	DepSet depset.DependencySet
```

Comment:
What's the relationship between `Cluster` and `ClusterConfig`? After a `Cluster` is created, the link between them is broken and only persists partially through the `DependencySet` - wouldn't it be worth it to expose `ClusterConfig()` on `Cluster`?

Also `DepSet` is abbreviated here but not above in `Cluster`

Reply:
`ClusterConfig` is to instantiate the default shim, with the values that make up the test object, including test handle and logger and such.
`DependencySet` is a protocol type, and one of the things we want to serve to the test via public interface.

And sure, will fix the naming